packages/veritone-widgets/src/widgets/EngineSelection/EngineListView/SelectBar/index.js

```diff
@@ -41,46 +46,45 @@ export default class SelectBar extends React.Component {
         />
         <div>Select All ({this.props.count})</div>
 
-        {!this.props.hideActionMenuItems && (
-          <div className={styles.selectBarIcons}>
-            <SearchBar
-              id={this.props.id}
-              onSearch={this.props.onSearch}
-              onClearSearch={this.props.onClearSearch}
-              onToggleSearch={this.props.onToggleSearch}
-              searchQuery={this.props.searchQuery}
-              isOpen={this.props.isSearchOpen}
-            />
-            {this.props.actionMenuItems && (
-              <div className={styles.actionMenuDividerContainer}>
-                <div className={styles.actionMenuDivider} />
-              </div>
-            )}
-            {this.props.actionMenuItems &&
-              this.props.actionMenuItems.map((item, key) => {
-                if (item.buttonText) {
-                  return (
-                    <Button
-                      key={key} // eslint-disable-line
-                      onClick={item.onClick}
-                      color="primary"
-                    >
-                      {item.buttonText}
-                    </Button>
-                  );
-                }
-
+        <div className={styles.selectBarIcons}>
+          <SearchBar
+            id={this.props.id}
+            currentTabIndex={this.props.currentTabIndex}
+            onSearch={this.props.onSearch}
+            onClearSearch={this.props.onClearSearch}
+            onToggleSearch={this.props.onToggleSearch}
+            searchQuery={this.props.searchQuery}
+            isOpen={this.props.isSearchOpen}
+          />
+          {this.props.actionMenuItems && (
+            <div className={styles.actionMenuDividerContainer}>
+              <div className={styles.actionMenuDivider} />
+            </div>
+          )}
+          {this.props.actionMenuItems &&
```

Comment:
try not to use `return` in these inner render functions (i wrote about this in the react standards doc)

also you shouldn't use index as a key... button text + icon class should be unique among items, right?

```javascript
{this.props.actionMenuItems &&
            this.props.actionMenuItems.map(
              ({ buttonText, iconClass, onClick }) =>
                buttonText ? (
                  <Button
                    key={`${buttonText}-${iconClass}`}
                    onClick={onClick}
                    color="primary"
                  >
                    {buttonText}
                  </Button>
                ) : (
                  <IconButton
                    key={`${buttonText}-${iconClass}`}
                    onClick={onClick}
                  >
                    <i className={iconClass} />
                  </IconButton>
                )
            )}
```